module/scripts/combat/applyCrit.js

```diff
@@ -8,25 +8,25 @@ export function addCritMessageContextOptions(html, options) {
             name: `${game.i18n.localize("WITCHER.Context.applyCritDmg")}`,
             icon: '<i class="fas fa-user-minus"></i>',
             condition: wasCritted,
-            callback: li => {
-                applyCritDamage(getInteractActor(), li[0].dataset.messageId)
+            callback:async li => {
+                applyCritDamage(await getInteractActor(), li[0].dataset.messageId)
```

Comment:
async callbacks are an anti pattern. You basically call an async (callback) function inside an async function. (https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns#the-deferred-anti-pattern). They also do not behave as expected depending on how/where they are used.

Can you not somehow make is sync again? Or maybe pass the actor ID to `applyCritDamage`, so that it can fetch it by his own. Maybe it is not necessary for the calling function to actually wait until the crit damage is applied, since it is a "simple" event callback?

Reply:
https://github.com/witchertrpg-foundryvtt/TheWitcherTRPG/issues/62